source/MaterialXGenShader/GenOptions.h

```diff
@@ -93,6 +93,12 @@ class GenOptions
     /// compress a set of UDIMs into a single normalized image for
     /// hardware rendering.
     bool hwNormalizeUdimTexCoords;
+
+    /// Variables for texture space renderings
+    bool textureSpaceRender;
+    std::string textureSpaceInput;
```

Comment:
Here I think it would be better to store the original InputPtr, which gives us access to both the input name and input type string.